core/src/domain/services/lease.service.ts

```diff
@@ -28,9 +30,10 @@ export class LeaseService {
         if (!estate) {
             throw new NotFoundException('Estate does not exist');
         }
-        const lease = LeaseDtoMapper.toModel(leaseDto, estate, agent, tenant);
-
-        return LeaseDtoMapper.fromModel(await this.leaseRepository.save(lease));
+        const lease = await this.leaseRepository.save(LeaseDtoMapper.toModel(leaseDto, estate, agent, tenant));
+        console.log('Lease created', lease);
+        this.eventEmitter.emit('lease.created', lease);
```

Comment:
and here you are giving a Lease and not LeaseInspectionCreationOnLeaseCreationEvent  to the handler so it must not work